dmt/main/models.py

```diff
@@ -1406,7 +1406,7 @@ def add_reply(self, user, body):
         self.email_reply(body, user, n)
 
     def email_reply(self, body, user, reply):
-        if self.author == user:
+        if self.user.userprofile == user:
```

Comment:
The intention is actually to eventually deal only in plain `User`s and only ever reference the `UserProfile`s when we need specific attributes from them. So, once more of these refactorings go through, the `user` passed in will be a `User` rather than a `UserProfile`.